### PR TITLE
Upgrade BookKeeper to 4.14.1

### PR DESCRIPTION
**Pull request by hangc0276**
### Motivation
Pulsar 2.8.0 should dependent on BookKeeper 4.14.0,  otherwise will cause direct memory copy to heap.  Discuss in https://github.com/apache/pulsar/pull/10330

This bug is fixed by https://github.com/apache/bookkeeper/pull/2701

### Modification
1. upgrade BookKeeper to 4.14.0
2. fix prometheus metric incompatible with bookkeeper 4.14.0, causing by interface changed by https://github.com/apache/bookkeeper/pull/2650

### CODE REVIEW

**Comment by hangc0276** [comment]
BookKeeper 4.14.0 has prometheus metric bug, which will cause BookKeeper prometheus metric can't be accepted by Prometheus. Please don't merge this PR before it addressed.

**Comment by eolivelli** [comment]
> BookKeeper 4.14.0 has prometheus metric bug, which will cause BookKeeper prometheus metric can't be accepted by Prometheus. Please don't merge this PR before it addressed.

Which bug?
Can you please link here the bug in BK issue tracker?

**Comment by hangc0276** [comment]
> > BookKeeper 4.14.0 has prometheus metric bug, which will cause BookKeeper prometheus metric can't be accepted by Prometheus. Please don't merge this PR before it addressed.
> 
> Which bug?
> Can you please link here the bug in BK issue tracker?

The BK bug is https://github.com/apache/bookkeeper/pull/2718 
Please help take a look before Pulsar 2.8.0 release. @sijie @jiazhai @merlimat @codelipenghui @eolivelli

**Comment by merlimat** [comment]
@hangc0276 The release 4.14.1 is already available on Maven central. You can update this PR.

**Comment by hangc0276** [comment]
@merlimat @eolivelli @codelipenghui  I have upgrade bookkeeper dependency to 4.14.1, PTAL, thanks.

File: pulsar-sql/presto-distribution/LICENSE [hunk 1]
```diff
@@ -413,18 +413,18 @@ The Apache Software License, Version 2.0
...
+    - bookkeeper-proto-4.14.0.jar
+    - bookkeeper-server-4.14.0.jar
+    - bookkeeper-stats-api-4.14.0.jar
+    - bookkeeper-tools-framework-4.14.0.jar
+    - circe-checksum-4.14.0.jar
+    - codahale-metrics-provider-4.14.0.jar
+    - cpu-affinity-4.14.0.jar
+    - http-server-4.14.0.jar
+    - prometheus-metrics-provider-4.14.0.jar
+    - codahale-metrics-provider-4.14.0.jar
```
Comment (codelipenghui) [inline pulsar-sql/presto-distribution/LICENSE:427]: Should also fixes the version in the LICENSE

File: distribution/server/src/assemble/LICENSE.bin.txt [hunk 2]
```diff
@@ -390,31 +390,31 @@ The Apache Software License, Version 2.0
...
+    - org.apache.bookkeeper-stream-storage-service-impl-4.14.0.jar
+    - org.apache.bookkeeper.http-http-server-4.14.0.jar
+    - org.apache.bookkeeper.http-vertx-http-server-4.14.0.jar
+    - org.apache.bookkeeper.stats-bookkeeper-stats-api-4.14.0.jar
+    - org.apache.bookkeeper.stats-prometheus-metrics-provider-4.14.0.jar
+    - org.apache.distributedlog-distributedlog-common-4.14.0.jar
+    - org.apache.distributedlog-distributedlog-core-4.14.0-tests.jar
+    - org.apache.distributedlog-distributedlog-core-4.14.0.jar
+    - org.apache.distributedlog-distributedlog-protocol-4.14.0.jar
+    - org.apache.bookkeeper.stats-codahale-metrics-provider-4.14.0.jar
```
Comment (codelipenghui) [inline distribution/server/src/assemble/LICENSE.bin.txt:417]: Should also fixes the version in the LICENSE

**Review by codelipenghui — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM

**Pull request merged**